libraries/AP_Math/SCurve.h

```diff
@@ -219,6 +220,11 @@ class SCurve {
         float end_pos;      // final position value for segment
     } segment[segments_max];
 
-    Vector3f track;       // total change in position from origin to destination
-    Vector3f delta_unit;  // reference direction vector for path
+    bool is_arc_segment;    // true if this segment is a circular arc, false if straight line
```

Comment:
if it was up to me, I probably would have put all the arc related variables (aka members) into a structure something like this:

struct {
   float angle_rad;
   length_ne;
} arc;